src/unitxt/llm_as_judge.py

```diff
@@ -121,22 +122,22 @@ def compute(
         )
 
         card = f"cards.dynamic_cards_for_llm_judges.{self.task}"
-        recipe = (
-            f"card={card},"
-            f"template={self.template},"
-            "demos_pool_size=0,"
-            "num_demos=0"
-        )
+        recipe_args = {
+            "card": card,
+            "template": self.template,
+            "demos_pool_size": 0,
+            "num_demos": 0,
+            "type": settings.default_recipe,
+        }
         if self.system_prompt:
-            recipe = f"{recipe},system_prompt={self.system_prompt}"
+            recipe_args["system_prompt"] = self.system_prompt
         if self.format:
-            recipe = f"{recipe},format={self.format}"
-
+            recipe_args["format"] = self.format
+        recipe = Artifact.from_dict(recipe_args)
         dataset = produce(instances, recipe)
         verdicts = self.inference_model.infer(dataset)
         meta_scores = evaluate(predictions=verdicts, data=dataset)
```

Comment:
Why is this necessary? Just for the postprocessors utilisation?

Reply:
The code here is using unitxt to build a dataset for the judge inference. The old code used a representation of string, that represents dict, that represents a recipe.  In addition to the fact that this way is much less clear and bug-prone, it doesn't allow to use ad-hoc template.

Reply:
Agree, the question is about the use of `evaluate`, is it just for the post-processing aspect?

Reply:
I believe so.  Right now, there is no simple way to run the post processors without executing the metrics.